pkg/templates/machinecontroller/deployment.go

```diff
@@ -917,5 +918,14 @@ func machineControllerDeployment(cluster *kubeoneapi.KubeOneCluster, credentials
 				},
 			},
 		},
-	}, nil
+	}
+
+	cabundle.Inject(cluster.CABundle, &dep.Spec.Template)
```

Comment:
The condition is checked inside the `Inject()`. This `Inject()` is followed by the `if cabundle != ""` only here. Other places are "blind"-ly injecting and caller doesn't need to account for conditions. So that's why it's like that. For consistency with other called.